sdk/core/Microsoft.Extensions.Azure/src/Internal/AzureClientFactory.cs

```diff
@@ -41,9 +45,12 @@ internal class AzureClientFactory<TClient, TOptions>: IAzureClientFactory<TClien
         public TClient CreateClient(string name)
         {
             _logForwarder.Start();
+
             if (!_clientRegistrations.TryGetValue(name, out ClientRegistration<TClient> registration))
             {
-                throw new InvalidOperationException($"Unable to find client registration with type '{typeof(TClient).Name}' and name '{name}'.");
+                _fallbackFactory ??= new FallbackAzureClientFactory<TClient>(_globalOptions, _serviceProvider, _logForwarder);
```

Comment:
Just curious - Is there a reason we defer the initialization of `FallbackAzureClientFactory` until `CreateClient` is called (vs building it in the constructor?)

Reply:
Yeah, there is quite a bit of reflection in FallbackAzureClientFactory ctor and I'd like to move a bit of those heave operations of the startup part especially considering it might not be needed at all.

Reply:
Makes sense!